tests/manage/z_cluster/test_osd_heap_profile.py

```diff
@@ -73,9 +75,39 @@ def test_osd_heap_profile(self):
                 osd_pod_profile = osd_pod
 
         log.info(f"Verify osd.{osd_id}.profile log exist on /var/log/ceph/")
-        out = osd_pod_profile.exec_cmd_on_pod(command="ls -ltr /var/log/ceph/")
-        log.info(out)
-        assert (
-            f"osd.{osd_id}.profile" in out
-        ), f"osd.{osd_id}.profile log does not exist on /var/log/ceph/\n{out}"
+        sample = TimeoutSampler(
+            timeout=100,
+            sleep=10,
+            func=self.verify_output_command_osd_pod,
+            command="ls -ltr /var/log/ceph/",
+            pod_obj=osd_pod_profile,
+            osd_id=osd_id,
+        )
+        if not sample.wait_for_func_status(result=True):
+            log.error(f"osd.{osd_id}.profile log does not exist on /var/log/ceph")
+            raise ValueError(
+                f"osd.{osd_id}.profile log does not exist on /var/log/ceph"
+            )
+
         log.info(f"osd.{osd_id}.profile log exist on /var/log/ceph")
+
+    def verify_output_command_osd_pod(self, command, pod_obj, osd_id):
```

Comment:
I think it will be clearer if you add another param `str_to_check`, or a similar name. Then you can pass the string "osd.{osd_id}.profile" in the test to the function `verify_output_command_osd_pod` instead of writing it twice.

Reply:
@yitzhak12 In the case of ERROR, I want to fail the test, and not use generator...

Reply:
But still, you can add the param `str_to_check` to the function. 
What I mean is to change the code to the following: 
`def verify_output_command_osd_pod(self, command, pod_obj, osd_id, str_to_check):`

And then:
`sample = TimeoutSampler(
            timeout=100,
            sleep=10,
            func=self.verify_output_command_osd_pod,
            command="ls -ltr /var/log/ceph/",
            pod_obj=osd_pod_profile,
            osd_id=osd_id,
            str_to_check="osd.{osd_id}.profile"
        )`

Reply:
done